torch/_dynamo/exc.py

```diff
@@ -115,6 +115,7 @@ def __init__(self, msg):
 
 class UserErrorType(Enum):
     DYNAMIC_CONTROL_FLOW = auto()
+    GRAPH_BREAK_IN_CONTROL_FLOW = auto()
```

Comment:
![image](https://github.com/pytorch/pytorch/assets/5652049/85319318-3f7a-4761-9135-5a1f033427c9)

This shouldn't be a UserErrorType, because it's not supported in eager mode either. The recommendation here is to add a new type of exception. I'm not sure it should subclass Unsupported, it should probably just subclass RuntimeError.

Naming: UncapturableHigherOrderOpError ?